pkg/kv/kvserver/loqrecovery/collect.go

```diff
@@ -62,3 +76,68 @@ func CollectReplicaInfo(
 	}
 	return loqrecoverypb.NodeReplicaInfo{Replicas: replicas}, nil
 }
+
+// checkUncommittedDescriptorsInRaftLog iterates over committed but not applied
+// part of raft log and searches for writes to range descriptor. If any value
+// referencing local range descriptor key is found then we have a potential
+// pending update.
+func checkUncommittedDescriptorsInRaftLog(
+	rangeID roachpb.RangeID,
+	rangeKey roachpb.RKey,
+	startIndex uint64,
+	endIndex uint64,
+	reader storage.Reader,
+) (bool, error) {
+	rangeDescKey := keys.RangeDescriptorKey(rangeKey)
+
+	key := keys.RaftLogKey(rangeID, startIndex)
+	endKey := keys.RaftLogKey(rangeID, endIndex)
+	iter := reader.NewMVCCIterator(storage.MVCCKeyIterKind, storage.IterOptions{
+		UpperBound: endKey,
+	})
+	defer iter.Close()
+
+	var meta enginepb.MVCCMetadata
+	var ent raftpb.Entry
+
+	iter.SeekGE(storage.MakeMVCCMetadataKey(key))
+	for ; ; iter.Next() {
+		if ok, err := iter.Valid(); err != nil || !ok {
+			return false, err
+		}
+		if err := protoutil.Unmarshal(iter.UnsafeValue(), &meta); err != nil {
+			return false, errors.Wrap(err, "unable to decode raft log MVCCMetadata")
+		}
+		if err := storage.MakeValue(meta).GetProto(&ent); err != nil {
+			return false, errors.Wrap(err, "unable to unmarshal raft Entry")
+		}
+		if ent.Type != raftpb.EntryNormal || len(ent.Data) == 0 {
```

Comment:
configuration changes come as an EntryConfChange, so you want to inspect those in particular. When you have an EntryConfChange(V1|V2), the command is nested like this:

https://github.com/cockroachdb/cockroach/blob/4df8ac262e72c58cb4e09dcf1beb0b8ff6fdde27/pkg/kv/kvserver/debug_print.go#L223-L243